File: Cap 06/6.2 - Matrizes de Uma Dimensao/Media Geral V3/medverv3.rb
=begin
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 * 
 * Capitulo ..: 6 - Topico 6.2
 * 
 * Programa ..: MEDIA GERAL - VERSAO 3
=end

puts 'MEDIA GERAL - V3'
puts

md = Array.new(8, 0.0)
soma = 0.0

for i in 1..8 do
  print "Entre a nota #{i}: "
  md[i] = gets.chomp.to_f
  soma += md[i]
end

media = soma / 8

for i in 1..8 do
  puts "A[#{i.to_s.rjust(2)}] = #{md[i].to_s.rjust(5, '0.0')}"
end

puts "Resultado da media = #{media.round(1)}"

File: Cap 06/6.2 - Matrizes de Uma Dimensao/Media Geral V3/medverv3.kt
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 6 - Topico 6.2
 * 
 * Programa ..: MEDIA GERAL - VERSAO 3
 * 
 */

import java.util.Scanner

fun main() {
  
    var MD = FloatArray(8)
    var SOMA = 0f
    var MEDIA: Float
    var entrada = Scanner(System.`in`)

    println("MEDIA GERAL - V3\n")

    for (i in 0..7) {
        print("Entre a nota ${i + 1}: ")
        MD[i] = entrada.nextFloat()
        SOMA += MD[i]
    }

    MEDIA = SOMA / 8

    for (i in 0..7) {
        println("A[$i] = %.1f".format(MD[i]))
    }

    println("Resultado da media = %.1f".format(MEDIA))
	
}
